src/TestSuite/TestCase.php

```diff
@@ -869,11 +869,14 @@ protected function _getTableClassName(string $alias, array $options): string
      * Set the app namespace
      *
      * @param string $appNamespace The app namespace, defaults to "TestApp".
-     * @return void
+     * @return string|null The previous app namespace or null if not set.
      */
-    public static function setAppNamespace(string $appNamespace = 'TestApp'): void
+    public static function setAppNamespace(string $appNamespace = 'TestApp'): ?string
     {
+        $previous = Configure::read('App.namespace');
         Configure::write('App.namespace', $appNamespace);
+
+        return $previous;
```

Comment:
How do you see this being used in our test suites or in an app's?

Reply:
I added an example conversion. There are a few cases for this. The reason I added it as return is this wraps the setter but you have to manually read the value if you want to save it which makes this wrapper pointless.

We have tests where the app namespace is set and never restored -- probably because nothing implies you can/should save it.

We have cases where the configure is manually queried to save when this wrapper is meant to hide that.

We have unit tests where the app namespace cannot be 'TestApp' which can happen accidentally depending on the order of execution because we don't have a clean save/restore pattern.